### ISSUE TITLE
App crashes due to possible thread-safety issues in onActivityResult

### ISSUE THREAD

**Issue by ragnor-rs** [miguelpruivo/flutter_file_picker#204]
**Describe the bug**
I have an app deployed to production with flutter_file_picker 1.4.3+2 installed. I get crash reports in Crashlytics.

**Issue details**
1. Android
2. 6.0.1, 7.0, 8.1.0, 9
3. FileType.IMAGE

**Error Log**
Stack trace 1:
```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke interface method 'void io.flutter.plugin.common.MethodChannel$Result.success(java.lang.Object)' on a null object reference
       at e.d.a.a.a.a$c.run(:8)
       at android.os.Handler.handleCallback(Handler.java:907)
       at android.os.Handler.dispatchMessage(Handler.java:105)
       at android.os.Looper.loop(Looper.java:216)
       at android.app.ActivityThread.main(ActivityThread.java:7625)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:524)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:987)
```

Stack trace 2:
```
Caused by java.lang.IllegalStateException: Reply already submitted
       at io.flutter.embedding.engine.dart.DartMessenger$Reply.reply(:35)
       at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler$1.success(:14)
       at e.d.a.a.a.a$a.onActivityResult(:37)
       at io.flutter.app.FlutterPluginRegistry.onActivityResult(:18)
       at io.flutter.app.FlutterActivityDelegate.onActivityResult(:6)
       at io.flutter.app.FlutterActivity.onActivityResult(:2)
       at android.app.Activity.dispatchActivityResult(Activity.java:7341)
       at android.app.ActivityThread.deliverResults(ActivityThread.java:4361)
       at android.app.ActivityThread.handleSendResult(ActivityThread.java:4409)
       at android.app.ActivityThread.-wrap19()
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1696)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:176)
       at android.app.ActivityThread.main(ActivityThread.java:6662)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:547)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:873)
```

**Flutter Version details**
```
[✓] Flutter (Channel stable, v1.12.13+hotfix.7, on Mac OS X 10.15.3 19D76, locale en-RU)
    • Flutter version 1.12.13+hotfix.7 at /Users/reist/Library/flutter
    • Framework revision 9f5ff2306b (2 weeks ago), 2020-01-26 22:38:26 -0800
    • Engine revision a67792536c
    • Dart version 2.7.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/reist/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3.1, Build version 11C504
    • CocoaPods version 1.8.4

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 43.0.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] Connected device (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 7.1.1 (API 25) (emulator)

• No issues found!
```

**Additional context**
I'm sure that it's flutter_file_picker because
- it's the only plugin which uses onActivityResult
- it's the only plugin which posts something to UI thread
- the problem occurs when an image is being uploaded (according to additional event logging)

I see a new thread is created on every onActivityResult call. This might be a problem.

Possibly related to #159. This is my top crash in Firebase (about 1.1% of users)

**Comment by miguelpruivo** [comment]
Hi, on which phone that happens?

**Comment by ragnor-rs** [comment]
@miguelpruivo 

**HUAWEI**
Honor 10
HUAWEI P smart Z
Mate 20 lite
P9 lite

**OPPO**
A1k

**Samsung**
Galaxy Note10+
Galaxy A7 (2018)
Galaxy A30s

**Xiaomi**
Redmi 7
Redmi 6
MIX 2S

**LeMobile**
Le 2

**Comment by miguelpruivo** [comment]
Are those rooted?

**Comment by miguelpruivo** [comment]
There have been some devices (not much, as you can see, 1%) that somehow have the `MethodChannel.result` as a `null` reference in a place where it can't ever be. Which somehow make me believe that those must be rooted devices or some roms that handle the Activity lifecycle in a different manner that they should, making it impossible for me to debug & fix.

**Comment by ragnor-rs** [comment]
Well, not rooted according to Firebase:

Stack trace 1:
<img width="383" alt="Screenshot 2020-02-12 at 21 21 18" src="https://user-images.githubusercontent.com/3357827/74364687-a9cb3700-4ddd-11ea-9467-18dbb4206b4e.png">

Stack trace 2:
<img width="380" alt="Screenshot 2020-02-12 at 21 21 05" src="https://user-images.githubusercontent.com/3357827/74364730-c5ced880-4ddd-11ea-9ed9-2585c7fa28e9.png">

**Comment by miguelpruivo** [comment]
This might be happening due to a leak on activity lifecycle. I'm already working on a V2 support that might get rid of it. I'll keep you updated. I'm closing this as the discussion related to it is on #195. 

Thank you!

**Comment by Marcosmaliki** [comment]
I still have this issue with the current version (1.5.0) on Huawei y9

**Comment by miguelpruivo** [comment]
@Marcosmaliki please post your logs about this.

There are some Android OEM Roms that break the activity protocols and make it somehow hard to handle/cover all the possibilities.

That exception could never happen because it shouldn’t be null where it does.

**Comment by miguelpruivo** [comment]
@Marcosmaliki could you please try with the temporary workaround that I added right now, please? 

Just replace it with: `file_picker: 1.5.0+1`. 

Basically, there is an issue with Flutter engine for V2 embedded that has only landed a fix on `master`, `dev` and `beta` channels.

**Comment by pmrajani** [comment]
Any news when this will fix? can't able to open File-picker on latest verson for Huawei  device.

**Comment by miguelpruivo** [comment]
Yes @pmrajani this should be fixed on 1.5.1 as it has already V2 embedded support.

Clean your project and update the dependency.

Thank you!
